src/plugin/cesium/tiles/cesium3dtileslayer.js

```diff
@@ -151,19 +189,66 @@ plugin.cesium.tiles.Layer.prototype.onTileProgress = function(pendingRequests, t
 plugin.cesium.tiles.Layer.prototype.restore = function(config) {
   plugin.cesium.tiles.Layer.base(this, 'restore', config);
 
+  const layerId = this.getId();
+  const accessToken = os.settings.get(`plugin.cesium.SettingsKey.ACCESS_TOKEN.${layerId}`);
+
   if (typeof config['assetId'] == 'number') {
     this.assetId = /** @type {number} */ (config['assetId']);
   }
 
   if (config['accessToken']) {
     this.accessToken = /** @type {string} */ (config['accessToken']);
+  } else {
+    this.accessToken = /** @type {string} */ (accessToken);
+  }
+
+  if (config['tileStyle']) {
+    this.tileStyle = /** @type {Object|string} */ (config['tileStyle']);
   }
 
   if (config['url']) {
     this.url = /** @type {string} */ (config['url']);
   }
 
   this.synchronize();
+
+  // Save the current layer so that we can run synchronize() on it asynchronously
+  const layerRef = this;
+  if (!this.accessToken) {
+    os.ui.window.launchConfirmText(/** @type {!osx.window.ConfirmTextOptions} */ ({
```

Comment:
This block might be useful extracted as a generic convenience function, like `plugin.cesium.getAccessToken`. The function should return a promise that resolves on confirm, rejects on cancel. There is an example of this in `os.track.promptForTitle`.